packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -328,11 +328,12 @@ export class RestApi extends Resource implements IRestApi {
   }
 
   /**
-   * Add an ApiKey
+   * Add an ApiKey, optionally with a supplied key value. If provided, the key value must be at least 20 characters.
    */
-  public addApiKey(id: string): IApiKey {
+  public addApiKey(id: string, value?: string): IApiKey {
```

Comment:
This change will make the API less extendible. 

Instead create a new interface called `ApiKeyOptions` which will be inherited into `ApiKeyProps`. The second parameter of this API call will be of the type `ApiKeyOptions`.

Here's an example - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L666-L682

Reply:
By doing this, and moving `apiKeyName` to `ApiKeyOptions`, you will be able to fix this as well - https://github.com/aws/aws-cdk/issues/7767 😊